pkgs/top-level/unix-tools.nix

```diff
@@ -0,0 +1,89 @@
+{ pkgs, buildEnv, runCommand, hostPlatform }:
+
+let
+
+  singleBinary = cmd: providers:
+    if builtins.hasAttr hostPlatform.parsed.kernel.name providers then
+      runCommand cmd {} ''
+        mkdir -p $out/bin
+
+        if ! [ -x "${providers.${hostPlatform.parsed.kernel.name}}/bin/${cmd}" ]; then
+          echo "Cannot find command ${cmd}"
+          exit 1
+        fi
+
+        ln -s ${providers.${hostPlatform.parsed.kernel.name}}/bin/${cmd} $out/bin/${cmd}
```

Comment:
@Ericson2314 Is there a better way to get this? Basically I want to get the second part of the "x86_64-darwin" tuple.

Reply:
Don't think so! haha. Maybe eventually `parsed` will contain everything and then we can remove the indirection.

Reply:
Could this be a normal `stdenv.derivation` with `meta.platforms` with regards to `borg` and other tooling? or is throw handled in the same way as `meta.platforms`?

Reply:
"throw" is allowed to be a little more descriptive which in this case seems good. I think in the end meta.platforms calls throw if system doesn't match.

Reply:
Ping me and I'll fix it after that is merged :).